metals/src/main/scala/scala/meta/internal/metals/BSPErrorHandler.scala

```diff
@@ -0,0 +1,172 @@
+package scala.meta.internal.metals
+
+import java.nio.file.Files
+import java.util.concurrent.atomic.AtomicReference
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.clients.language.MetalsLanguageClient
+import scala.meta.io.AbsolutePath
+
+import org.eclipse.{lsp4j => l}
+
+class BSPErrorHandler(
+    languageClient: MetalsLanguageClient,
+    workspaceFolder: AbsolutePath,
+)(implicit context: ExecutionContext) {
+
+  private def logsPath = workspaceFolder.resolve(Directories.log)
+  private val currentError =
+    new AtomicReference[CurrentAndPreviousError](new CurrentAndPreviousError())
+
+  def onError(message: String): Unit = {
+    val collectedError = currentError.getAndUpdate(_.append(message))
+    if (collectedError.isEmpty) {
+      showErrorWithDelay()
+    }
+  }
+
+  private def showErrorWithDelay() = {
+    Future {
+      Thread.sleep(1000) // 1 sec
+      currentError.getAndUpdate(_.newError())
+    }.flatMap { error =>
+      error.error match {
+        case Some(message) => showError(filerANSIColorCodes(message))
+        case None => Future.unit
+      }
+    }
+  }
+
+  private def showError(message: String): Future[Unit] = {
+    scribe.error(s"${BSPErrorHandler.errorInBsp}\n$message")
```

Comment:
Could we instead of looking for it later save the current line number here?

Reply:
I don't see how. We need the line number in the log file. The output of scribe is redirected, so I don't see how scribe would have that info.

Reply:
I was thinking to just open the file and count lines before sending an error?

Reply:
I mean, in theory something might get written in between the actions.

Reply:
But again it won't probably be super common and only probably up to a few lines off...